presto-docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -0,0 +1,197 @@
+
+In this example, the table is partitioned by month and by a hash of the account number::
+
+    CREATE TABLE iceberg.testdb.sample_partitioned (
+        order_date DATE,
+        account_number BIGINT,
+        customer VARCHAR)
+    WITH (partitioning = ARRAY['month(order_date)', 'bucket(account_number, 10)'])
+
+Deletion by Partition
+---------------------
+
+For partitioned tables, the Iceberg connector supports the deletion of entire
+partitions if the WHERE clause specifies an identity transform of a partition
+column.  Given this table definition::
+
+    CREATE TABLE test_partition_deletion (
+        c integer,
+        d double)
+        WITH (partitioning = ARRAY['c'])
+
+This SQL will delete all partitions for which c is 1::
+
+    DELETE FROM test_partition_deletion
+    WHERE c = 1
+
+The current Iceberg connector is based on Iceberg v1.  Iceberg v2 is not yet
+released.  When v2 is released, it will support row-level deletion and the
+Iceberg connector will be updated to provide row-level deletion.
+
+The Iceberg connector currently supports only deletion by partition.
+This SQL below will fail because the doesn't span entire partitions::
+
+    DELETE FROM test_partition_deletion
+    WHERE c = 1 AND d > 10
+
+Rolling Back to a Previous Snapshot
+-----------------------------------
+
+Iceberg supports a "snapshot" model of data, where table snapshots are
+identified by an ``long`` snapshot_ids.
+
+The connector provides a system snapshots table for each Iceberg table.  Snapshots are
+identified by BIGINT snapshot ids.  You can find the latest snapshot id for table
+foo by running the following command::
+
+    SELECT snapshot_id FROM 'foo$snapshots' ORDER BY committed_at DESC LIMIT 1
+
+A SQL procedure ``system.rollback_to_snapshot`` allows the caller to roll back
+the state of the table to a previous snapshot id::
+
+    CALL system.rollback_to_snapshot(schema_name, table_name, snapshot_id)
+
+Schema Evolution
+----------------
+
+Iceberg and the Iceberg connector support schema evolution, with safe
+column add, drop, reorder and rename operations, including in nested structures.
+Table partitioning can also be changed and the connector can still
+query data created before the partitioning change.
+
+Migrating Existing Tables
+-------------------------
+
+The connector can read from or write to Hive tables that have been migrated to Iceberg.
+Currently, there is no Presto support to migrate Hive tables to Presto, so you will
+need to use either the Iceberg API or Spark.
+
+System Tables and Columns
+-------------------------
+
+The connector supports ``table$partitions`` as a substitute for Hive's ``SHOW PARTITIONS``.
+The differences are that it returns some partition metrics for each partition value
+and you can also use it on unpartitioned tables.
+
+Iceberg supports ``$snapshot_id`` and ``$snapshot_timestamp_ms`` as hidden columns.
```

Comment:
Does this work? We don't have any code or tests for this.

Reply:
The connector should support table names like `abc@123` or `abc$partitions@456` but I can't find any tests for this. It might just work -- we should add tests.

Reply:
I'll make a note to add the tests.

Reply:
> Does this work? We don't have any code or tests for this.

I added a test for this in AbstractTestIcebergSmoke, now disabled, because AFAICT these columns do not exist.  We should decide if they should exist, and if so add them and add more tests for them.